devito/finite_differences/differentiable.py

```diff
@@ -144,24 +198,46 @@ def __eq__(self, other):
         return super(Differentiable, self).__eq__(other) and\
             all(getattr(self, i, None) == getattr(other, i, None) for i in self._state)
 
+    def index(self, dim):
+        inds = [self.dimensions[i] for i, d in enumerate(self.dimensions) if d == dim]
+        return inds[0]
+
+    @property
+    def name(self):
+        return "".join(f.name for f in self._functions)
+
     @property
     def laplace(self):
         """
         Generates a symbolic expression for the Laplacian, the second
         derivative w.r.t all spatial Dimensions.
         """
-        space_dims = [d for d in self.indices if d.is_Space]
+        space_dims = [d for d in self.dimensions if d.is_Space]
         derivs = tuple('d%s2' % d.name for d in space_dims)
         return Add(*[getattr(self, d) for d in derivs])
 
+    @property
+    def div(self):
+        space_dims = [d for d in self.indices if d.is_Space]
```

Comment:
Is `self.indices` already filtered in some way so we avoid, in certain cases, ending up with repetitions of the same dimension?